SalesforceCommerceCloud/pwa-kit
# @W-12627224@ Account menu a11y

This PR contains a couple of a11y fixes for the account dropdown

* Using a header instead of `<p>` tag on the account drop down menu header
* Using box to indicate current selection in addition to color. This change also affects other `menu-link` items.
* Using underline to indicate hover in addition to color. This change also affects other `menu-link` items.
* Wrapping the menu items in a `<li>` tag 

Desktop view changes: (Order History is active page, Account Details is hovered)
![Screenshot 2024-07-08 at 5 16 49 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47404250/f2d8c283-8ec3-48c7-a179-b2e567d6d333)

Mobile view changes:  (Wishlist is active page, Order History is hovered)
![Screenshot 2024-07-08 at 5 28 01 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47404250/40c74ec3-4abf-4ddc-91d7-2a2a52e9eb23)

Testing these changes:
* Pull down these changes and start the app
* Log in
* Expand the 'My Account' drop down menu
* Verify menu header is `<h2>` instead of `<p>`. Also verify that the text styling is the same as before
* Verify each menu item is wrapped in a `<li>`
* Verify that hovering over menu items now underlines the selection
* Click a menu item
* Verify the selected page is now indicated in the drop down menu as well as the left account page nav
